File: backend/app/services/auth_service.py
from fastapi import HTTPException
from sqlalchemy.orm import Session
from app.models.user import User
from app.schemas.user import UserCreate
from app.core.security import hash_password, verify_password
import uuid
from datetime import datetime, timezone


def create_user(db: Session, user: UserCreate):
    # Check if email already exists
    existing_user = db.query(User).filter(User.email == user.email).first()
    if existing_user:
        raise HTTPException(status_code=400, detail="Email already registered")

    hashed = hash_password(user.password)

    db_user = User(
        id=uuid.uuid4(),
        name=user.name,
        email=user.email,
        password_hash=hashed,
        created_at=datetime.now(timezone.utc)
    )

    db.add(db_user)
    db.commit()
    db.refresh(db_user)

    return db_user


def authenticate_user(db: Session, email: str, password: str):
    user = db.query(User).filter(User.email == email).first()

    if not user:
        return None

    if not verify_password(password, user.password_hash):
        return None

    return user

File: backend/src/services/auth.service.js
const prisma = require('../database/client');
const { hashPassword, verifyPassword } = require('../utils/hashing');

async function createUser(data) {
  const existing = await prisma.user.findUnique({ where: { email: data.email } });
  if (existing) {
    const err = new Error('Email already registered');
    err.status = 400;
    throw err;
  }
  const password_hash = await hashPassword(data.password);
  return prisma.user.create({
    data: { name: data.name, email: data.email, password_hash },
  });
}

async function authenticateUser(email, password) {
  const user = await prisma.user.findUnique({ where: { email } });
  if (!user) return null;
  const valid = await verifyPassword(password, user.password_hash);
  if (!valid) return null;
  return user;
}

module.exports = { createUser, authenticateUser };
